### PR TITLE
Write unit tests for API Examples of checkbox.0, checkbox.1

### PR DESCRIPTION
**Pull request by Moelfarri**
Write unit tests for API Examples of checkbox.0, checkbox.1

Part of  #130459 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: examples/api/test/material/checkbox/checkbox.1_test.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Checkbox can be checked', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.CheckboxExampleApp(),
+    );
+
+
```
Comment (goderbauer) [inline examples/api/test/material/checkbox/checkbox.1_test.dart:16]: nit: remove one of the blank lines?

File: examples/api/test/material/checkbox/checkbox.1_test.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  checkbox = tester.widget(find.byType(Checkbox).first);
+  checkboxWithError = tester.widget(find.byType(Checkbox).at(1));
+  checkboxDisabled = tester.widget(find.byType(Checkbox).last);
+
+  expect(checkbox.value, isFalse);
+  expect(checkboxWithError.value, isFalse);
+  expect(checkboxDisabled.value, isFalse);
+});
+}
```
Comment (goderbauer) [inline examples/api/test/material/checkbox/checkbox.1_test.dart:70]: The indentation in this file seems off. I believe line 17 and lines 22-69 need to be indented by two extra spaces?

File: examples/api/test/material/checkbox/checkbox.1_test.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  await tester.pump();
+  checkbox = tester.widget(find.byType(Checkbox).first);
+  checkboxWithError = tester.widget(find.byType(Checkbox).at(1));
+  checkboxDisabled = tester.widget(find.byType(Checkbox).last);
+
+  expect(checkbox.value, isNull);
+  expect(checkboxWithError.value, isNull);
+  expect(checkboxDisabled.value, isNull);
+
+  /// Tap the second Checkbox and verify the state change.
```
Comment (goderbauer) [inline examples/api/test/material/checkbox/checkbox.1_test.dart:46]:
```suggestion
  // Tap the second Checkbox and verify the state change.
```

File: examples/api/test/material/checkbox/checkbox.1_test.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  checkbox = tester.widget(find.byType(Checkbox).first);
+  checkboxWithError = tester.widget(find.byType(Checkbox).at(1));
+  checkboxDisabled = tester.widget(find.byType(Checkbox).last);
+
+  expect(checkbox.value, isFalse);
+  expect(checkboxWithError.value, isFalse);
+  expect(checkboxDisabled.value, isFalse);
+
+  /// Tap the third Checkbox and verify that should remain unchanged.
```
Comment (goderbauer) [inline examples/api/test/material/checkbox/checkbox.1_test.dart:58]:
```suggestion
  // Tap the third Checkbox and verify that should remain unchanged.
```

File: examples/api/test/material/checkbox/checkbox.1_test.dart [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  expect(checkboxDisabled.value, isTrue);
+
+  expect(checkboxWithError.isError, isTrue);
+  expect(checkboxDisabled.onChanged, null);
+
+  expect(checkbox.tristate, isTrue);
+  expect(checkboxWithError.tristate, isTrue);
+  expect(checkboxDisabled.tristate, isTrue);
+
+
```
Comment (goderbauer) [inline examples/api/test/material/checkbox/checkbox.1_test.dart:34]: nit: remove one blank line

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks!

**Pull request merged**